server/src/entity/user.entity.ts

```diff
@@ -18,8 +19,12 @@ export class User {
   @Column('text')
   email: string;
 
+  @OneToOne(type=> Ranger, ranger => ranger.ranger)
```

Comment:
What happens when the user isn't a ranger? And I don't know if we ever want to or need to find the ranger row through a user?

Is it possible to do one-to-one mapping from only the one direction?

Reply:
I fixed it so that only user that are rangers can be added.
The purpose of finding the ranger in the user table is to make sure only rangers are in the ranger table.

I believe so, since the user table is not concerned about what is in the ranger table and the ranger tables' rangers will be linked to one user in the user table.